apps/bfd-server/dev/bfd-data-dictionary/app/dd-transformer/dd_to_csv.py

```diff
@@ -44,39 +44,37 @@ def build_csv_row(element):
 parser.add_argument('-m', '--template', dest='template', type=str,  required=True, help='template file name to use e.g. ./template/v2-to-csv.json')
 args = parser.parse_args()
 
+template = Path(args.template)
+source_dir = Path(args.source)
+target_path = Path(args.target)
+
 # open files
-with open(args.template, 'r') as template_file:
+with open(template, 'r') as template_file:
     template_json = json.load(template_file)
-template_file.close()
-out_file = open(args.target, 'w', newline='')
-csv_writer = csv.writer(out_file)
 
 print("Generating CSV from DD content")
 print("")
-print("Template:  " + args.template)
-print("Source:    " + args.source)
-print("Target:    " + args.target)
+print("Template:  " + str(template))
+print("Source:    " + str(source_dir))
+print("Target:    " + str(target_path))
 
 # main loop
-csv_writer.writerow(build_csv_header())
-ct = 0
-for file_name in os.listdir(args.source):    # loop thru source data folder, read each json file, translate to csv and write row to file
-    if file_name.endswith(".json"):
-        source_path = args.source + "\\" + file_name
-        with open(source_path) as element_file:
-            element_json = json.load(element_file)
-            row = build_csv_row(element_json)
-            csv_writer.writerow(row)
-            element_file.close()
-            ct += 1
+#
+with open(target_path, 'w', newline='') as out_file:
+    csv_writer = csv.writer(out_file)
+    csv_writer.writerow(build_csv_header())
+    ct = 0
+    for file_name in source_dir.iterdir():    # loop thru source data folder, read each json file, translate to csv and write row to file
+        try:
+            with open(file_name) as element_file:
+                element_json = json.load(element_file)
+                row = build_csv_row(element_json)
+                csv_writer.writerow(row)
```

Comment:
Is this performant enough? I know Pandas is pretty fast if this doesnt whip through the data fast enough, but Im not sure if the number of rows is enough to bother.

Reply:
the current DD is < 650 rows, and will probably never exceed 700.  So I believe that performance/scale won't be an issue.  The current scripts typically run in < 1 sec.  Having said that, I would not at all be opposed to incorporating a library like Pandas.